pkg/controller/seed-controller-manager/constraint-controller/controller.go

```diff
@@ -235,19 +257,34 @@ func (r *reconciler) reconcile(ctx context.Context, constraint *kubermaticv1.Con
 		}
 	}
 
-	// constraint creation
+	if err = r.cleanupConstraint(ctx, log, constraint, unwantedClusterList); err != nil {
+		return err
+	}
+
+	if err = r.createConstraint(ctx, log, constraint, clusterList); err != nil {
```

Comment:
I would use `ensure` rather than `create` as it does not always imply a creation.